src/common/ContractInteractions.ts

```diff
@@ -0,0 +1,35 @@
+import TransferRequest from "./common-types";
+import {ProvidersRegistryState} from "../providers-registry/types";
+import {TokenState} from "../token/types";
+
+declare const SmartWeave;
+
+export const registryTxId = "XQkGzXG6YknJyy-YbakEZvQKAWkW2_aPRhc3ShC8lyA";
+
+/**
+ * this class simplifies interactions between RedStone SmartWeave contracts
+ */
+export class ContractInteractions {
+
+  static async providerContractState(): Promise<ProvidersRegistryState> {
+    const providersRegistryTxId = await ContractInteractions.getContractTxId("providers-registry");
+    return await SmartWeave.contracts.readContractState(providersRegistryTxId);
+  }
+
+  static async tokenContractState(): Promise<TokenState> {
+    const providersRegistryTxId = await ContractInteractions.getContractTxId("token");
+    return await SmartWeave.contracts.readContractState(providersRegistryTxId);
+  }
+
+  static async generateId(request: any): Promise<string> {
+    const stakeRequestBuffer = SmartWeave.arweave.utils.stringToBuffer(request);
+    const hash = await SmartWeave.arweave.crypto.hash(stakeRequestBuffer, "SHA-256");
```

Comment:
Just in case I'll add sth random like current block height just to make it harder to generate hash collision in case a user is free to define all the request parameters.

Reply:
there is already Smartweave.block.timestamp...